### DISCUSSION TITLE
Adding text at home assistant sensor

### DISCUSSION THREAD

**Discussion by diego065** [pixelit-project/PixelIt#141] [Node-RED Flows]
Hello everyone!
I would like to add some custom text to the sensor value that is displayed.. any idea on how to do that?
To be clear, I have a sensor showing the temperature after the 8x8 bitmap but I also would like to add something like "External {value from HA} °C "... until now I couldn't find any way to show on the display all the information I want.
I tried as the attached picture but it only shows me one info!
Thanks in advance
<img width="764" alt="Schermata 2022-02-15 alle 12 14 51" src="https://user-images.githubusercontent.com/7717688/154051420-246a250d-001d-4502-ac0a-49c4a303b3d4.png">

**Comment by o0shojo0o** [comment]
Hi, 
I always do this with a "Function Node".
![image](https://user-images.githubusercontent.com/19333515/154069210-15051ff3-06e4-4c40-b743-3fc121aef5cb.png)
This could then look like this:
![image](https://user-images.githubusercontent.com/19333515/154069740-7a92b300-7f06-4c62-9d17-2b50db89dc3f.png)
But there is also the syntax (which I also prefer):
![image](https://user-images.githubusercontent.com/19333515/154071078-9adc1a78-0f9b-4e62-91a3-fbc5dc5c3b62.png)

**Reply by diego065** [reply]
Thanks! What's the differences between the 2? It will be also possible to "combine" two different attributes or sensor?

**Reply by aritzurruela** [reply]
Also interested in combining two variables in one screen.

**Reply by o0shojo0o** [reply]
I usually don't do this in NodeRed, but in my case in ioBroker.

**Reply by ojarosch** [reply]
The values of two different sensors could be combined like this:
![image](https://user-images.githubusercontent.com/9081883/186686200-ba450616-e2e8-4a1c-8d50-15ce333471ef.png)

```json
[{"id":"c4132efe91eeaf68","type":"inject","z":"a9932bcf1db982ed","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"30","crontab":"","once":false,"onceDelay":"0,1","topic":"","payload":"","payloadType":"date","x":150,"y":80,"wires":[["22e1feab934f8c26"]]},{"id":"22e1feab934f8c26","type":"api-current-state","z":"a9932bcf1db982ed","name":"","server":"6b1fc163.10009","version":3,"outputs":1,"halt_if":"","halt_if_type":"str","halt_if_compare":"is","entity_id":"sensor.temperature_wetterstation_balkon","state_type":"str","blockInputOverrides":false,"outputProperties":[{"property":"temp","propertyType":"msg","value":"","valueType":"entityState"}],"for":"0","forType":"num","forUnits":"minutes","override_topic":false,"state_location":"payload","override_payload":"msg","entity_location":"data","override_data":"msg","x":470,"y":80,"wires":[["28f0f00d43d49258"]]},{"id":"1ca8885cc2e0e944","type":"debug","z":"a9932bcf1db982ed","name":"debug 1","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":500,"y":200,"wires":[]},{"id":"28f0f00d43d49258","type":"api-current-state","z":"a9932bcf1db982ed","name":"","server":"6b1fc163.10009","version":3,"outputs":1,"halt_if":"","halt_if_type":"str","halt_if_compare":"is","entity_id":"sensor.humidity_wetterstation_balkon","state_type":"str","blockInputOverrides":false,"outputProperties":[{"property":"humid","propertyType":"msg","value":"","valueType":"entityState"}],"for":"0","forType":"num","forUnits":"minutes","override_topic":false,"state_location":"payload","override_payload":"msg","entity_location":"data","override_data":"msg","x":470,"y":140,"wires":[["d910d3cab965072c"]]},{"id":"d910d3cab965072c","type":"function","z":"a9932bcf1db982ed","name":"function 1","func":"msg.payload = `${msg.temp}°C ${msg.humid}%`;\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":360,"y":200,"wires":[["1ca8885cc2e0e944"]]},{"id":"6b1fc163.10009","type":"server","name":"Home Assistant","version":4,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":false,"heartbeat":false,"heartbeatInterval":"30","areaSelector":"friendlyName","deviceSelector":"friendlyName","entitySelector":"friendlyName","statusSeparator":"at: ","statusYear":"hidden","statusMonth":"short","statusDay":"numeric","statusHourCycle":"h23","statusTimeFormat":"h:m"}]
```

Content of `Function1` Node:
```js
msg.payload = `${msg.temp}°C ${msg.humid}%`;
return msg;
```

---

The difference between:

```js
msg.payload = "External " + msg.payload + " °C"
```
and


```js
msg.payload = `External ${msg.payload}°C`
```

Is that the first one uses string concatination and the second one uses a template literal.
When writing clean code, the second one is prefered.

String concatination can, in some edge cases, result in wierd outputs.
